cli/cook/subcommands/submit.py

```diff
@@ -0,0 +1,186 @@
+    """
+    for cluster in clusters:
+        cluster_name = cluster['name']
+        try:
+            url = make_url(cluster, 'rawscheduler')
+            print_info('Attempting to submit on %s cluster...' % colors.bold(cluster_name))
+            resp = http.post(url, json={'jobs': jobs})
+            logging.info('response from cook: %s' % resp.text)
+            print_submit_result(cluster, resp)
+            if resp.status_code == requests.codes.created:
+                return 0
+        except requests.exceptions.ConnectionError as ce:
+            logging.info(ce)
+            reason = 'Cannot connect to %s (%s)' % (cluster_name, cluster['url'])
+            print_info('%s\n' % submit_failed_message(cluster_name, reason))
+    raise Exception(colors.failed('Job submission failed on all of your configured clusters.'))
+
+
+def read_commands_from_stdin():
+    """Prompts for and then reads subcommands, one per line, from stdin"""
+    print_info('Enter the subcommands, one per line (press Ctrl+D on a blank line to submit)')
+    commands = read_lines()
+    if len(commands) < 1:
+        raise Exception('You must specify at least one command')
+    return commands
+
+
+def read_jobs_from_stdin():
+    """Prompts for and then reads job(s) JSON from stdin"""
+    print('Enter the raw job(s) JSON (press Ctrl+D on a blank line to submit)', file=sys.stderr)
+    jobs_json = sys.stdin.read()
+    return jobs_json
+
+
+def submit(clusters, args):
+    """
+    Submits a job (or multiple jobs) to cook scheduler. Assembles a list of jobs,
+    potentially getting data from configuration, the command line, and stdin.
+    """
+    logging.debug('submit args: %s' % args)
+
+    job = args
+    raw = job.pop('raw', None)
+    command_from_command_line = job.pop('command', None)
+
+    if raw:
+        if command_from_command_line:
+            raise Exception('You cannot specify a command at the command line when using --raw/-r')
+
+        jobs_json = read_jobs_from_stdin()
+        jobs = parse_raw_job_spec(job, jobs_json)
+    else:
+        if command_from_command_line:
+            if command_from_command_line[0] == '--':
+                command_from_command_line = command_from_command_line[1:]
+            commands = [' '.join(command_from_command_line)]
+        else:
+            commands = read_commands_from_stdin()
+
+        logging.debug('subcommands: %s' % commands)
+
+        if job.get('uuid') and len(commands) > 1:
+            raise Exception('You cannot specify multiple subcommands with a single UUID')
+
+        if job.get('env'):
+            job['env'] = dict([e.split('=', maxsplit=1) for e in job['env']])
+
+        jobs = [deep_merge(job, {'command': c}) for c in commands]
+
+    for j in jobs:
+        if not j.get('uuid'):
+            j['uuid'] = str(uuid.uuid4())
+
+        if not j.get('name'):
+            j['name'] = '%s_job' % os.environ['USER']
```

Comment:
Is this conventional or something you came up with?

Reply:
Our internal `cook-submit` tool uses `"cookjob"` as the default name. We can stay consistent with that, but I think I like the idea of the user being in the default name.

Reply:
OK